### PR TITLE
Create updated Feast Serving service.proto for Feast 0.3

### PR DESCRIPTION
**Pull request by woop**

### CODE REVIEW

File: protos/feast/serving/ServingService.proto [hunk 1]
```diff
@@ -19,55 +19,140 @@ syntax = "proto3";
...
-    rpc QueryFeatures (QueryFeaturesRequest) returns (QueryFeaturesResponse) {
-    };
-
     // Retrieve version information about this Feast deployment
-    rpc GetFeastServingVersion (GetFeastServingVersionRequest) returns (GetFeastServingVersionResponse);
-}
+    rpc GetFeastServingVersion (google.protobuf.Empty) returns (GetFeastServingVersionResponse);
+
+    // Get Feast serving storage type (online or offline)
+    rpc GetFeastServingType (google.protobuf.Empty) returns (GetFeastServingTypeResponse);
```
Comment (pradithya) [inline protos/feast/serving/ServingService.proto]: So a serving service will only serve 1 serving type?

File: protos/feast/serving/ServingService.proto [hunk 2]
```diff
@@ -19,55 +19,140 @@ syntax = "proto3";
...
+
+    // Get online features from Feast serving
+    rpc GetOnlineFeatures (GetFeaturesRequest) returns (GetOnlineFeaturesResponse);
```
Comment (zhilingc) [inline protos/feast/serving/ServingService.proto]: The methods for online and offline requests are different?

File: protos/feast/serving/ServingService.proto [hunk 3]
```diff
@@ -19,55 +19,140 @@ syntax = "proto3";
...
+
+    // Get offline features from Feast serving. This is an async job.
+    rpc GetOfflineFeatures (GetFeaturesRequest) returns (GetOfflineFeaturesResponse);
+
+    // Get the current status of an offline feature request job
+    rpc GetOfflineFeaturesJobStatus (OfflineFeaturesJob.GetStatusRequest) returns (OfflineFeaturesJob.GetStatusResponse);
```
Comment (zhilingc) [inline protos/feast/serving/ServingService.proto]: Does this mean that serving has to maintain some state about what jobs are running and where for offline feature gets?

File: protos/feast/serving/ServingService.proto [hunk 4]
```diff
@@ -19,55 +19,140 @@ syntax = "proto3";
...
+    repeated FeatureId featureIds = 1;
+    UserEntityData userEntityData = 2;
+
+    message FeatureId {
+        string featureSetName = 1;
+        string featureSetVersion = 2;
+        string featureName = 3;
+    }
+
+    message UserEntityData {
```
Comment (zhilingc) [inline protos/feast/serving/ServingService.proto]: UserEntity might be a word that is misconstrued if you plan on adding entities back in, imo

Comment (woop) [reply protos/feast/serving/ServingService.proto]: Are you able to come up with a better name?

File: protos/feast/serving/ServingService.proto (hunk 3 above)
Comment (woop) [reply protos/feast/serving/ServingService.proto]: Yes

File: protos/feast/serving/ServingService.proto (hunk 2 above)
Comment (woop) [reply protos/feast/serving/ServingService.proto]: Yes. We can either have one rpc with a flag or two rpcs. This seems to be more explicit and will be visible in stack traces.

File: protos/feast/serving/ServingService.proto (hunk 1 above)
Comment (woop) [reply protos/feast/serving/ServingService.proto]: For now yes.

File: protos/feast/serving/ServingService.proto [hunk 5]
```diff
@@ -19,55 +19,140 @@ syntax = "proto3";
...
 }
 
-message QueryFeaturesResponse {
-    // Entity name of the response
-    string entityName = 1;
-    // map of entity ID and its entity's properties.
-    map<string, Entity> entities = 2;
+message GetFeaturesRequest {
+    repeated FeatureId featureIds = 1;
+    UserEntityData userEntityData = 2;
```
Comment (pradithya) [inline protos/feast/serving/ServingService.proto]: What would it represents? Is it the replacement of entity Ids?

File: protos/feast/serving/ServingService.proto [hunk 6]
```diff
@@ -19,55 +19,140 @@ syntax = "proto3";
...
+        string name = 1;
+        feast.types.ValueType valueType = 2;
+    }
+
+    message UserEntityRow {
+        repeated feast.types.Value Value = 1;
+    }
+
+    // Defines relationships to join user entity data to feature set data
+    message JoinOn {
```
Comment (pradithya) [inline protos/feast/serving/ServingService.proto]: Is it a field in GetFeaturesRequest?

File: protos/feast/serving/ServingService.proto (hunk 3 above)
Comment (woop) [reply protos/feast/serving/ServingService.proto]: But in theory we could keep this in memory if we dont mind losing it if serving goes down.

File: protos/feast/serving/ServingService.proto (hunk 5 above)
Comment (woop) [reply protos/feast/serving/ServingService.proto]: It encapsulates the entity ids and event time

**Pull request merged**